libs/shared/src/hooks/use-staking-actions/use-staking-actions.ts

```diff
@@ -491,6 +1016,65 @@ export function useStakingActions() {
     ],
   );
 
+  // Estimate fee for precompile claiming rewards from a single validator
+  const handlePrecompileClaimRewardEstimatedFee = useCallback(
+    async (validatorAddress: string): Promise<EstimatedFeeResponse> => {
+      if (!ethAddress || !config) {
+        throw new Error('Insufficient data for fee estimation');
+      }
+
+      try {
+        const encodedData = encodeFunctionData({
+          abi: withdrawDelegatorRewardAbi,
+          functionName: 'withdrawDelegatorReward',
+          args: [ethAddress, validatorAddress],
+        });
+
+        console.log('Encoded data:', encodedData);
+
+        const estimatedGas = await estimateGas(config, {
+          to: STAKING_PRECOMPILE_ADDRESS,
```

Comment:
Incorrect precompile address. Rewards are being handled by Distribution precompile.
ABI for claiming single validator rewards
```
{
    "inputs": [
      {
        "internalType": "address",
        "name": "delegatorAddress",
        "type": "address"
      },
      {
        "internalType": "string",
        "name": "validatorAddress",
        "type": "string"
      }
    ],
    "name": "withdrawDelegatorRewards",
    "outputs": [
      {
        "components": [
          {
            "internalType": "string",
            "name": "denom",
            "type": "string"
          },
          {
            "internalType": "uint256",
            "name": "amount",
            "type": "uint256"
          }
        ],
        "internalType": "struct Coin[]",
        "name": "amount",
        "type": "tuple[]"
      }
    ],
    "stateMutability": "nonpayable",
    "type": "function"
  },
```